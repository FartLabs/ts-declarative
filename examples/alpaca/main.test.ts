import { assertSnapshot } from "@std/testing/snapshot";
import { jsonSchemaPerson, renderTemplate } from "./main.ts";

Deno.test("Generate Alpaca form from JSON Schema", async (t) => {
  await assertSnapshot(
    t,
    await renderTemplate(JSON.stringify(jsonSchemaPerson)),
  );
});
