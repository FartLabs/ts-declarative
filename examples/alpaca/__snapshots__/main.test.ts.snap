export const snapshot = {};

snapshot[`Generate Alpaca form from JSON Schema 1`] = `
'<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Alpaca with @fartlabs/declarative</title>

    <!-- dependencies (jquery, handlebars and bootstrap) -->
    <script
      type="text/javascript"
      src="//code.jquery.com/jquery-1.11.1.min.js"
    ></script>
    <script
      type="text/javascript"
      src="//cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.5/handlebars.min.js"
    ></script>
    <link
      type="text/css"
      href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap.min.css"
      rel="stylesheet"
    />
    <script
      type="text/javascript"
      src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.1/js/bootstrap.min.js"
    ></script>

    <!-- alpaca -->
    <link
      type="text/css"
      href="//cdn.jsdelivr.net/npm/alpaca@1.5.27/dist/alpaca/bootstrap/alpaca.min.css"
      rel="stylesheet"
    />
    <script
      type="text/javascript"
      src="//cdn.jsdelivr.net/npm/alpaca@1.5.27/dist/alpaca/bootstrap/alpaca.min.js"
    ></script>
  </head>
  <body>
    <div id="form1"></div>
    <script type="text/javascript">
      \$("#form1").alpaca({ "schema": {"type":"object","properties":{"givenName":{"type":"string","title":"givenName"},"familyName":{"type":"string","title":"familyName"}},"required":["givenName","familyName"]} })
    </script>
  </body>
</html>
'
`;
